src/contracts/atlas/AtlasVerification.sol

```diff
@@ -381,39 +384,85 @@ contract AtlasVerification is EIP712, DAppIntegration, AtlasConstants {
         return ValidCallsResult.Valid;
     }
 
+    /// @notice The _handleUserNonces internal function handles the verification of user nonces for both sequential and
+    /// non-sequential nonce systems.
+    /// @param user The address of the user to verify the nonce for.
+    /// @param nonce The nonce to verify.
+    /// @param sequential A boolean indicating if the nonce mode is sequential (true) or not (false)
+    /// @param isSimulation A boolean indicating if the execution is a simulation.
+    /// @return validNonce A boolean indicating if the nonce is valid.
+    function _handleUserNonces(
+        address user,
+        uint256 nonce,
+        bool sequential,
+        bool isSimulation
+    )
+        internal
+        returns (bool validNonce)
+    {
+        NonceTracker memory nonceTracker = userNonceTrackers[user];
+        (validNonce, nonceTracker) = _handleNonces(nonceTracker, user, true, nonce, sequential);
```

Comment:
I think we can use the same pass-by-struct-reference syntax here as used in this PR https://github.com/FastLane-Labs/atlas/pull/256 which means we don't have to explicitly return the `nonceTracker` struct.
```suggestion
        validNonce = _handleNonces(nonceTracker, user, true, nonce, sequential);
```